### PR TITLE
#1261 Added collecting parameter design pattern

### PR DESCRIPTION
**Pull request by JoshuaSinglaANU**
Pull request description
Added the code, tests, and README.md for the collecting parameter design pattern.

Was unable to get the class diagram to generate.

### CODE REVIEW

**Comment by JoshuaSinglaANU** [comment]
UML reverse wrapper added, yet still no class diagram generation.

**Comment by JoshuaSinglaANU** [comment]
UML diagram added.

File: collecting-parameter/README.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
+---
+layout: pattern
+title: Collecting Parameter
+folder: collecting-parameter
+permalink: /patterns/collecting-parameter/
+categories: Structural
+language: en
+tags:
+- Collecting Parameter
+---
```
[the remark this answers is not in the document]
Comment (iluwatar) [reply collecting-parameter/README.md:10]: I suggest category `Idiom` and tags `Generic`. See https://github.com/iluwatar/java-design-patterns/wiki/07.-Categories-and-Tags

File: collecting-parameter/README.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+## Name / classification
```
Comment (iluwatar) [inline collecting-parameter/README.md:12]:
```suggestion
## Name
```

File: collecting-parameter/README.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+### In plain words
+Instead of having one giant method that contains numerous policies for collecting information into a variable, we can
+create numerous smaller functions that each take parameter, and append new information. We can pass the parameter to
+all of these smaller functions and by the end, we will have what we wanted originally. This time, the code is cleaner
+and easier to understand. Because the larger function has been broken down, the code is also easier to modify as changes
+are localised to the smaller functions.
+
+### Wikipedia says
+In the CollectingParameter idiom a collection (list, map, etc.) is passed repeatedly as a parameter to a method which adds items to the collection.
```
Comment (iluwatar) [inline collecting-parameter/README.md:32]:
```suggestion
In the Collecting Parameter idiom a collection (list, map, etc.) is passed repeatedly as a parameter to a method which adds items to the collection.
```

File: collecting-parameter/pom.xml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                </configuration>
+                <executions>
+                    <execution>
+                        <phase>process-classes</phase>
+                        <goals>
+                            <goal>map</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (iluwatar) [inline collecting-parameter/pom.xml:87]: This part is already defined in the parent pom.xml and shouldn't normally need any additional configuration here

File: collecting-parameter/src/main/java/com/iluwatar/collectingparameter/App.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+      **/
+    printerQueue.addPrinterItem(new PrinterItem(PaperSizes.A4, 5, false, false));
+    printerQueue.addPrinterItem(new PrinterItem(PaperSizes.A3, 2, false, false));
+    printerQueue.addPrinterItem(new PrinterItem(PaperSizes.A2, 5, false, false));
+
+    var result = new LinkedList<PrinterItem>();
+
+    addValidA4Papers(result);
+    addValidA3Papers(result);
+    addValidA2Papers(result);
```
Comment (iluwatar) [inline collecting-parameter/src/main/java/com/iluwatar/collectingparameter/App.java:68]: Add comments to this part as well

**Review by iluwatar — CHANGES_REQUESTED** [review]
Looks good already. Added a few improvement suggestions.

File: collecting-parameter/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoshuaSinglaANU) [reply collecting-parameter/README.md]: Fixed now.

File: collecting-parameter/README.md (hunk 2 above)
Comment (JoshuaSinglaANU) [reply collecting-parameter/README.md]: Fixed now.

File: collecting-parameter/README.md (hunk 3 above)
Comment (JoshuaSinglaANU) [reply collecting-parameter/README.md]: Fixed now.

File: collecting-parameter/pom.xml (hunk 4 above)
Comment (JoshuaSinglaANU) [reply collecting-parameter/pom.xml]: Fixed now.

File: collecting-parameter/src/main/java/com/iluwatar/collectingparameter/App.java (hunk 5 above)
Comment (JoshuaSinglaANU) [reply collecting-parameter/src/main/java/com/iluwatar/collectingparameter/App.java]: Fixed now.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Excellent work! Thank you for the new pattern @JoshuaSinglaANU 💪

@all-contributors please add @JoshuaSinglaANU for code

**Pull request merged**